lib/puppet/provider/oneview_hypervisor_cluster_profile/c7000.rb

```diff
@@ -24,6 +24,50 @@
 
   mk_resource_methods
 
+  def exists?
+    create
+  end
+
+  def create
+    set_uri if resource['data'] && resource['data'].key?('hypervisorManagerUri')
+    @data = resource['data']
+    return true if resource_update
+    super(:create)
+  end
+
+  def set_uri
+    hm_class = OneviewSDK.resource_named('HypervisorManager', @client.api_version)
+    hm_uri = hm_class.find_by(@client, name: 'HM').first['uri']
```

Comment:
Why did you hard code the name as 'HM'. Usually hypervisor manager name refers to ip address